src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -276,10 +472,161 @@ public FileModel createContent(FileModel fileModel) throws DataPreparationExcept
+            JSONObject entryValueMap = (JSONObject) entryResponse.get("entry");
+
+            fileModel.setNodeRef(entryValueMap.get("id").toString());
+            fileModel.setName(entryValueMap.get("name").toString());
+            fileModel.setFileType(FileType.fromName(fileModel.getName()));
+
+            String fileLocation = Utility.buildPath(getLastResource(), newContent);
+            fileModel.setCmisLocation(fileLocation);
+            fileModel.setProtocolLocation(fileLocation);
+            fileModel.setContent("This is a test file");
+
+            logger.info(String.format("Successful created content with id '%s' ", entryValueMap.get("id").toString()));
+            return fileModel;
+        }
+        else
+        {
+            logger.error(client.getParameterFromJSON(response,"briefSummary", "error"));
+            return new FileModel();
+        }
+    }
+
+    /**
+     * This method updates the content of the created document
+     *
+     * @param client
+     * @param fileModel
+     * @param documentType
+     */
+    public void updateContent(AlfrescoHttpClient client, FileModel fileModel, DocumentType documentType)
+    {
+        // Build request
+        String nodeId = fileModel.getNodeRef();
+        String reqUrl = client.getApiVersionUrl() + "nodes/" + nodeId + "/content?majorVersion=true";
+
+        HttpPut put  = new HttpPut(reqUrl);
+        String fileContent = fileModel.getContent();
+        String contentType = documentType.type + ";charset=" + client.UTF_8_ENCODING;
+        put.addHeader("Content-Type", contentType);
+        StringEntity se = new StringEntity(fileContent, client.UTF_8_ENCODING);
+        se.setContentType(new BasicHeader(HTTP.CONTENT_TYPE, client.MIME_TYPE_JSON));
+        put.setEntity(se);
+
+        try
+        {
+            // Send Request
+            logger.info(String.format("Update content for file with name '%s' by: ", fileModel.getName()));
+            logger.info(String.format("POST: '%s'", reqUrl));
+            HttpResponse response = client.executeAndRelease(currentUser.getUsername(), currentUser.getPassword(), put);
+            if(HttpStatus.SC_OK == response.getStatusLine().getStatusCode())
+            {
+                logger.info(String.format("Successfully updated content"));
+            }
+        }
+        finally
+        {
+            put.releaseConnection();
+            client.close();
+        }
+    }
+
+    /**
+     * Creates a random document based on {@link DocumentType} passed using CMIS
+     * Return the {@link Document} object on success creation
+     * <code>
+     * dataContent.usingUser(userModel).usingResource(myFolder).createContent(DocumentType.TEXT_PLAIN);
+     * </code>
+     *
+     * @param documentType
+     * @return
+     * @throws DataPreparationException
+     */
+    public FileModel createContentDocTypeCmisApi(DocumentType documentType) throws DataPreparationException

```

Comment:
could this method be/contain duplicate code of createContentCmisApi ? refactor

Reply:
These look quite different to me, but there were other methods with this problem and I've changed those